Fix stride printing/parsing formatting

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45156 Fix stride printing/parsing formatting**

Differential Revision: [D24078695](https://our.internmc.facebook.com/intern/diff/D24078695)

Commit:
Fix stride printing/parsing formatting

[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

gmagogsfm commented on torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:
```diff
@@ -37,22 +37,23 @@ at::optional<Node*> FindFusibleListUnpack(Node* n) {
-//       4:3, 3:1) = aten::split_with_sizes(%input, %13, %7, %8)
-//   return (%14, %15, %16)
+//   %14 : Float(2, 4, 3, strides=[12, 3, 1]), %15 : Float(1, 4, 3, strides=[12,
+//   3, 1]), %16 : Float(2, 4, 3, strides=[12, 3, 1] =
```
Indent line 54 and 55

gmagogsfm commented on torch/onnx/__init__.py:
```diff
@@ -212,13 +212,13 @@ def export(model, args, f, export_params=True, verbose=False, training=TrainingM
+            The external binary files will be stored in the same location specified by the model
             location 'f'. If False, then the model is stored in regular format, i.e. model and
             parameters are all in one file. This argument is ignored for all export types other
-            than ONNX. 
```
It seems this file is still auto-formatted, so you may want to check you editor setting. But I think these space changes are small enough that we can include them in this PR.

ansleyadelaide replied:
Good call. I had Editor: Format on Save off already, but I turned Editor > Comments: Insert Space off as well, which I believe should fix the problem going forward.

gmagogsfm commented on aten/src/ATen/core/type.cpp:
```diff
@@ -41,10 +41,20 @@ std::ostream& operator<<(std::ostream & out, const Type & t) {
-            type_verbosity() >= TypeVerbosity::TypeAndStride) {
-          out << ":" << *value->strides()[i];
+      }
+      if (has_valid_strides_info) {
```
I might be missing something. Why is the condition here `has_valid_strides_info`? I thought the final `,` be printed as long as `*ndim > 0` regardless of `strides` condition.

gmagogsfm replied:
How about the case where there is complete strides info, but `type_verbosity` is lower than `TypeAndStride`? is `,` needed?

gmagogsfm replied:
Is this addressed?

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"




[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"


Differential Revision: [D24078695](https://our.internmc.facebook.com/intern/diff/D24078695)

[ghstack-poisoned]

Commit:
Update on "Fix stride printing/parsing formatting"


Differential Revision: [D24078695](https://our.internmc.facebook.com/intern/diff/D24078695)

[ghstack-poisoned]

ZolotukhinM commented:
Wow, nice, thanks for fixing that!
